simp_le.py

```diff
@@ -1435,6 +1435,26 @@ def finalize_order(client, order):
     return finalized_order
 
 
+def poll_and_answer(client, authorizations, roots):
+    """Poll authorization status and answer challenge if required"""
+    # pylint: disable=no-member
+    for name, auth in six.iteritems(authorizations):
+        for _ in range(5):
+            auth, _ = client.poll(auth)
+            if auth.body.status == messages.STATUS_VALID:
+                break
+            if auth.body.status == messages.STATUS_PENDING:
+                challb = supported_challb(auth)
+                response, validation = challb.response_and_validation(
+                    client.net.key)
+                save_validation(roots[name], challb, validation)
+                client.answer_challenge(challb, response)
+                break
+            time.sleep(1)
+        else:
+            raise Error('Authorization status timed out.')
```

Comment:
I moved polling and answering to its own function to avoiding piling up `# pylint: disable`, let me know if this works for you.

The tests are passing : https://travis-ci.com/github/buchdag/simp_le/builds/155895298